flink-core/src/main/java/org/apache/flink/types/parser/IntParser.java

```diff
@@ -37,7 +37,20 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, I
 		boolean neg = false;
 
 		final int delimLimit = limit-delimiter.length+1;
-
+		
+		int delimCount = 0;
+		for (int i = 0; i < delimiter.length; i++) {
+			if (bytes[startPos + i] == delimiter[i]) {
+				delimCount++;
+			} else {
+				break;
+			}
+		}
+		if (delimCount == delimiter.length) {
+			setErrorState(ParseErrorState.EMPTY_STRING);
+			return -1;
+		}
+		
```

Comment:
Could you explain why you added this loop and the check afterwards? There is already a function `delimiterNext` which checks if the delimiter comes next. I would suggest to integrate your check in the loop below which already checks for the delimiter using this method. You just need to add an additional condition if no data has been read and throw the error there.